src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -638,5 +639,145 @@ private DeclaredTypeAssignment GetModuleType(ModuleDeclarationSyntax syntax)
         // references to symbols can be involved in cycles
         // we should not try to obtain the declared type for such symbols because we will likely never finish
         private bool IsCycleFree(DeclaredSymbol declaredSymbol) => this.binder.TryGetCycle(declaredSymbol) is null;
+
+        /// <summary>
+        /// Returns the declared type of the resource body (based on the type string).
+        /// Returns the same value for single resource or resource loops declarations.
+        /// </summary>
+        /// <param name="resourceTypeProvider">resource type provider</param>
+        private static TypeSymbol GetDeclaredResourceType(IBinder binder, ResourceDeclarationSyntax resource)
```

Comment:
>GetDeclaredResourceType [](http://example.com/codeflow?start=34&length=23)

Was this just moved?

Reply:
Yeah - no changes. Next refactor I'm going to need to start modifying the logic.